### PR TITLE
fix(otellogs): configure imagePullSecrets for otellogs daemonset

### PR DESCRIPTION
**Pull request by kkujawa-sumo**
Pull request was tested manually with following configuration:
```yaml
sumologic:
  accessId: dummy
  accessKey: dummy
  endpoint: http://receiver-mock.receiver-mock:3000/terraform/api/
  pullSecrets:
    - name: container-secret-test

otellogs:
  image:
    repository: ghcr.io/kkujawa-sumo/sumologic-otel-collector
    tag: "0.73.0-sumo-1"
  daemonset:
    initContainers:
      changeowner:
        image:
          repository: ghcr.io/kkujawa-sumo/busybox
          tag: 1.36.0
          pullPolicy: IfNotPresent
```

new ServiceAccount:
```
$ kubectl get ServiceAccounts -n sumologic collection-sumologic-otelcol-logs-collector -o yaml
apiVersion: v1
imagePullSecrets:
- name: container-secret-test
kind: ServiceAccount
metadata:
  annotations:
    meta.helm.sh/release-name: collection
    meta.helm.sh/release-namespace: sumologic
  creationTimestamp: "2023-04-14T09:46:26Z"
  labels:
    app: collection-sumologic-otelcol-logs-collector
    app.kubernetes.io/managed-by: Helm
    chart: sumologic-3.4.0
    heritage: Helm
    release: collection
  name: collection-sumologic-otelcol-logs-collector
  namespace: sumologic
  resourceVersion: "105439"
  selfLink: /api/v1/namespaces/sumologic/serviceaccounts/collection-sumologic-otelcol-logs-collector
  uid: 6c223de8-88ee-4e4e-ad30-1694d8da42e8
secrets:
- name: collection-sumologic-otelcol-logs-collector-token-98q7n
```

### CODE REVIEW

**Comment by swiatekm-sumo** [comment]
Shouldn't we just use the common `imagePullSecrets` setting everything else uses instead of adding a new one?

**Review by swiatekm-sumo — APPROVED** [review]

File: deploy/helm/sumologic/templates/logs/collector/otelcol/daemonset.yaml [hunk 1]
```diff
@@ -67,6 +67,10 @@ spec:
       {{- else }}
       priorityClassName: {{ include "sumologic.metadata.name.priorityclass" $ctx | quote }}
       {{- end }}
+{{- if $.Values.sumologic.pullSecrets }}
```
Comment (rnishtala-sumo) [inline deploy/helm/sumologic/templates/logs/collector/otelcol/daemonset.yaml:70]: does the service account for the otelcol logs collector also need this option? Like here: https://github.com/SumoLogic/sumologic-kubernetes-collection/blob/main/deploy/helm/sumologic/templates/serviceaccount.yaml

Comment (kkujawa-sumo) [reply deploy/helm/sumologic/templates/logs/collector/otelcol/daemonset.yaml:70]: we use different `serviceaccount` for `collection-sumologic-otelcol-logs-collector` https://github.com/SumoLogic/sumologic-kubernetes-collection/blob/main/deploy/helm/sumologic/templates/logs/collector/otelcol/serviceaccount.yaml
I can move adding  `imagePullSecrets` for to the `serviceaccount` to have consistency

**Review by sumo-drosiek — APPROVED** [review]
I like the simplicty!

**Pull request merged**